clarity/src/vm/mod.rs

```diff
@@ -312,6 +332,15 @@ pub fn eval(
         Atom, AtomValue, Field, List, LiteralValue, TraitReference,
     };
 
+    if check_max_execution_time_expired(env.global_context) {
+        warn!(
+            "ExecutionTime expired while running {:?} ({:?} elapsed)",
+            exp,
+            env.global_context.execution_time_tracker.elapsed()
+        );
+        return Err(CheckErrors::ExecutionTimeExpired.into());
```

Comment:
If this returns an error, and this new error is not handled specially anywhere, then this is consensus breaking. I think this design needs some re-working. Let's chat tomorrow.